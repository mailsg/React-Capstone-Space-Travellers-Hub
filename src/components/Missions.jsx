import React, { useEffect } from 'react';
import Table from 'react-bootstrap/Table';
import { useDispatch, useSelector } from 'react-redux';
import { fetchMissions } from '../redux/missions/missionsSlice';

const Missions = () => {
  const dispatch = useDispatch();
  const missionsFetched = useSelector((state) => state.missions.missions.length > 0);

  useEffect(() => {
    if (!missionsFetched) {
      dispatch(fetchMissions());
    }
  }, [dispatch, missionsFetched]);

  const missionsList = useSelector((state) => state.missions.missions);

  return (
    <div className="mx-5">
      <Table striped bordered hover responsive="sm">
        <thead>
          <tr className="fw-bold">
            <th style={{ width: '10%' }}>Mission</th>
            <th style={{ width: '70%' }}>Description</th>
            <th style={{ width: '10%' }}>Status</th>
            <th style={{ width: '10%' }}>&nbsp;</th>
          </tr>
        </thead>
        <tbody>
          {missionsList.map((mission) => (
            <tr key={mission.mission_id}>
              <td className="fw-bold">{mission.mission_name}</td>
              <td>{mission.description}</td>
              <td>to be</td>
              <td>to be</td>
            </tr>
          ))}
        </tbody>
      </Table>
    </div>
  );
};

export default Missions;
