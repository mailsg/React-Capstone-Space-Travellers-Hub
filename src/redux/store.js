import { configureStore } from '@reduxjs/toolkit';
import rocketsSlice from './rockets/rocketSlice';

const store = configureStore({
  reducer: {
    rockets: rocketsSlice,
  },
});

export default store;
