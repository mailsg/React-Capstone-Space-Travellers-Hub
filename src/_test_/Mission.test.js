import '@testing-library/jest-dom';
import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import Missions from '../components/Missions';
import store from '../redux/store';

describe('User Interactions test', () => {
  it('will test user interaction', async () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <Missions />
        </Provider>,
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
