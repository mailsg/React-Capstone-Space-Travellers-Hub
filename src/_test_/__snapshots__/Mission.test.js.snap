// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User Interactions test will test user interaction 1`] = `
<div
  className="mx-5 px-4"
>
  <div
    className="table-responsive-sm"
  >
    <table
      className="table table-striped table-bordered table-hover"
    >
      <thead>
        <tr
          className="fw-bold"
        >
          <th
            style={
              Object {
                "width": "10%",
              }
            }
          >
            Mission
          </th>
          <th
            style={
              Object {
                "width": "60%",
              }
            }
          >
            Description
          </th>
          <th
            style={
              Object {
                "width": "15%",
              }
            }
          >
            Status
          </th>
          <th
            style={
              Object {
                "width": "15%",
              }
            }
          >
            Â 
          </th>
        </tr>
      </thead>
      <tbody />
    </table>
  </div>
</div>
`;
